_drgn.pyi

```diff
@@ -733,6 +733,41 @@ class Program:
         :param lang: :attr:`Type.language`
         """
         ...
+    def threads(self) -> ThreadIterator:
+        """
+        Return an iterable over all of the threads resident within the process.
+        """
+        ...
+    def thread(self, tid: int) -> Thread:
+        """
+        Retrieve the thread with the given `tid`, raising a `LookupError` in the
+        event no such thread exists.
+        """
+        ...
+    def crashed_thread(self) -> Thread:
+        """
+        Return the thread responsible for causing the process to crash (e.g. the thread that segfaulted,
+        received a `SIGQUIT`, etc.). This method should only be called when debugging a core dump.
```

Comment:
I would think that `DRGN_ERROR_OTHER` is a better fit here since it's not truly an invalid argument that will trigger the exception, but I'll defer to your judgement.

Reply:
Also right now much of the gating that checks that we are checking the correct target is in the form of `assert` statements present in the libdrgn internals. Is the intention to migrate those to returning errors triggering python exceptions?

Reply:
Yeah, `DRGN_ERROR_INVALID_ARGUMENT`/`ValueError` isn't perfect here, since it's `self` that's invalid, but I think it's close enough and more informative than a generic `Exception`.

Reply:
The existing assertions in this area check that libdrgn itself is making internal calls correctly in terms of how we store the PRSTATUS notes for the kernel vs userspace. For any user-facing API, we should return an error rather than asserting.